api/base/exceptions.py

```diff
@@ -210,6 +210,12 @@ class TwoFactorRequiredError(AuthenticationFailed):
     default_detail = _('Must specify two-factor authentication OTP code.')
     pass
 
+
 class InvalidModelValueError(JSONAPIException):
     status_code = 400
     default_detail = _('Invalid value in POST/PUT/PATCH request.')
+
+
+class InvalidTargetError(JSONAPIException):
```

Comment:
I don't think should this be a `JSONAPIException`/DRF exception. This is raised in the case of a programmer error, not a client input error. Therefore, the app should fail loudly, i.e. return a 500. This can just be a subclass of `Exception`.